rfcs/0084-ZHF-on-master-and-freeze-dates.md

```diff
@@ -0,0 +1,104 @@
+of forcing many, largely-untested, commits into master before branch-off. This
+happened for the 20.09 release in which ~640 staging commits were pushed through in the 
+days leading up to the branch off (https://github.com/NixOS/nixpkgs/pull/95492, 
+https://github.com/NixOS/nixpkgs/pull/96280, https://github.com/NixOS/nixpkgs/pull/96437,
+https://github.com/NixOS/nixpkgs/pull/97146). The last PR included a release blocker
+fix (introduced in [one of the mentioned PRs](https://github.com/NixOS/nixpkgs/pull/96437))
+which delayed the branch-off from 4 Sept 2020 to 7 Sept 2020, and ZHF had to
+be delayed another day to get a semi-accurate hydra build status. The systemd bump
+was also largely responsible delaying the release until after the scheduled release date
+where basic login features in plasma [weren't restored until Oct 5th, a week after the planned release date](https://github.com/NixOS/nixpkgs/pull/99629)
+and a backlog of other plasma related issues [weren't resolved until Oct 19th, almost a month after the planned release date](https://github.com/NixOS/nixpkgs/pull/101078).
+This is not to discredit any individual, but demonstrate that certain packages
+have many dimensions of compatibility which need a longer time to stabilize
+on unstable before being included in a release.
+
+## Core changes
+
+Staging-next iterations will change to one week.
+
+Updates to `staging` which will land before branch-off will be restricted to non-breaking updates, and will need to be more thoroughly tested. The below table is a more detailed timeline, it is also sanitized of events not relating to merge criteria, staging events, or ZHF.
+
+| Weeks from Release | Events |
+| --- | --- |
+| -8 Weeks | Gnome, Plasma(YY.11) packaging begins |
+| -6 Weeks | Freeze "critical packages" breaking updates |
+| -4 Weeks | Freeze staging, allow only non-breaking updates and Desktop Manager changes |
+| -3 Weeks | (Day before ZHF) merge staging-next, prep for ZHF |
+| -3 Weeks | Begin ZHF on master branch |
+| -3 Weeks | Strong emphasis on minimizing regressions in master PRs |
+| -2 Weeks | Merge first staging-next fixes, begin second staging-next fix cycle |
+| -2 Weeks | Unfreeze "critical packages" changes to staging |
+| -2 Weeks | Unfreeze staging, allow for normal staging workflow, these changes will not be present in master before branch-off |
+| -2 Weeks | `staging-next` remains fixes only, will be merged before branch-off |
+| -1 Weeks | Merge second staging-next fix cycle |
+| -1 Weeks | Perform Branch-off, create release channels, create new beta / unstable tags |
+| -1 Weeks | ZHF transitions to "backporting" workflow |
+| -1 Weeks | Prepare for release, finish remaining issues |
+| 0 Weeks | Release! |
+| 0 Weeks | ZHF Ends |
+
+Actual release schedule may adjust dates slightly due to unforeseen events.
+
+Release team will retain the right to refine what constitutes "critical packages", as this may
```

Comment:
What does "Release team" mean? I know that a few people (including you IIRC) plan to "define" such a team rather than having two RMs with most of the workload, but I'm not sure if it's clear who's part of this at the moment.

Reply:
Currently it means the release managers.

Reply:
I would like to eventually create a "release team", as many people have expressed interest in being a part of a release; but it's much bigger of a ask to be "the release manager".